File: app/src/main/java/com/hpc/vkd/DownloaderListener.java
package com.hpc.vkd;

import java.io.File;

/**
 * Class for listening downloader events through view
 */

interface DownloaderListener {

    void onFileNameReceived(String url, FileName fileName);

    void onLoadStarted(int notificationId, FileName fileName);

    void onLoadFinished(int notificationId, File file);
}

File: app/src/main/java/com/hpc/vkd/DownloaderListener.kt
package com.hpc.vkd

import java.io.File

/**
 * Class for listening downloader events through view
 */

interface DownloaderListener {

    fun onFileNameReceived(url: String, fileName: FileName)

    fun onLoadStarted(notificationId: Int, fileName: FileName)

    fun onLoadFinished(notificationId: Int, file: File)
}
